### PR TITLE
test: Add various low-level p2p tests

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK: More coverage is better

File: test/functional/p2p_invalid_tx.py [hunk 1]
```diff
@@ -148,6 +149,22 @@ def run_test(self):
...
+            orphan_tx_pool[i].vin.append(CTxIn(outpoint=COutPoint(i, 333)))
+            orphan_tx_pool[i].vout.append(CTxOut(nValue=11 * COIN, scriptPubKey=SCRIPT_PUB_KEY_OP_TRUE))
+
+        with node.assert_debug_log(['mapOrphan overflow, removed 1 tx']):
+            node.p2p.send_txs_and_test(orphan_tx_pool, node, success=False)
+
+        rejected_parent = CTransaction()
+        rejected_parent.vin.append(CTxIn(outpoint=COutPoint(tx_orphan_2_invalid.sha256, 0)))
+        rejected_parent.vout.append(CTxOut(nValue=11 * COIN, scriptPubKey=SCRIPT_PUB_KEY_OP_TRUE))
+        rejected_parent.rehash()
```
Comment (theStack) [inline test/functional/p2p_invalid_tx.py:164]: nit: could also use the method `calc_sha256()` here to be consistent with all other uses above

**Review by theStack — COMMENTED** [review]
Concept ACK

**Review by jonatack — COMMENTED** [review]
ACK fa4c29bc1d242

File: test/functional/p2p_invalid_tx.py (hunk 1 above)
Comment (MarcoFalke) [reply test/functional/p2p_invalid_tx.py:164]: I uesed `calc_sha256` when later `tx.sha256` was accessed and `rehash` when later `tx.hash` was accessed

File: test/functional/p2p_filter.py [hunk 2]
```diff
@@ -60,8 +60,13 @@ def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()
 
     def run_test(self):
-        self.log.info('Add filtered P2P connection to the node')
         filter_node = self.nodes[0].add_p2p_connection(FilterNode())
+
+        self.log.info('Check that too large filter is rejected')
+        with self.nodes[0].assert_debug_log(['Misbehaving']):
+            filter_node.send_and_ping(msg_filterload(data=b'\xaa', nHashFuncs=51, nTweak=0, nFlags=1))
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply test/functional/p2p_filter.py:67]: Will leave as-is for now to not invalidate reviews. Can be done as a follow-up

**Pull request merged**